File: app/src/main/java/com/tutv/android/ui/search/SearchView.java
package com.tutv.android.ui.search;

import com.tutv.android.domain.Genre;
import com.tutv.android.domain.Network;

import java.util.List;

public interface SearchView {
    void setSearchQuery(String searchQuery, Integer genre, Integer network);
    void setFilters(List<Genre> genres, List<Network> networks);
    void showFilterError();
}

File: app/src/main/java/com/tutv/android/ui/search/SearchView.kt
package com.tutv.android.ui.search

import com.tutv.android.domain.Genre
import com.tutv.android.domain.Network

interface SearchView {
    fun setSearchQuery(searchQuery: String?, genre: Int?, network: Int?)
    fun setFilters(genres: MutableList<Genre?>, networks: MutableList<Network?>)
    fun showFilterError()
}